src/Panel/Panel.php

```diff
@@ -45,7 +200,290 @@ public static function customCss(App $kirby)
+                'error'  => $message,
+                'layout' => Panel::hasAccess($kirby->user()) ? 'inside' : 'outside'
+            ],
+            'title' => 'Error'
+        ];
+    }
+
+    /**
+     * Creates $fiber response array
+     *
+     * @param \Kirby\Cms\App $kirby
+     * @param string $component
+     * @param array $data
+     * @return array
+     */
+    public static function fiber(App $kirby, string $component, array $data = []): array
+    {
+        // get all data for the request
+        $data = static::data($kirby, $data);
+
+        // filter data, if it's a partial request
+        $data = static::partial($kirby, $component, $data);
+
+        // resolve lazy data entries
+        return A::apply($data);
+    }
+
+    /**
+     * Returns the file object for the given
+     * parent path and filename
+     *
+     * @param string|null $path Path to file's parent model
+     * @param string $filename Filename
+     * @return \Kirby\Cms\File|null
+     * @throws \Kirby\Exception\NotFoundException if the file cannot be found
+     */
+    public static function file(string $path = null, string $filename)
+    {
+        $filename = urldecode($filename);
+        $file     = static::parent($path)->file($filename);
+
+        if ($file && $file->isReadable() === true) {
+            return $file;
+        }
+
+        throw new NotFoundException([
+            'key'  => 'file.notFound',
+            'data' => [
+                'filename' => $filename
+            ]
+        ]);
+    }
+
+    /**
+     * Check for access permissions
+     *
+     * @param \Kirby\Cms\User|null $user
+     * @param string|null $areaId
+     * @return bool
+     */
+    public static function firewall(?User $user = null, ?string $areaId = null): bool
+    {
+        // a user has to be logged in
+        if ($user === null) {
+            throw new PermissionException(t('error.access.panel'));
+        }
+
+        // get all access permissions for the user role
+        $permissions = $user->role()->permissions()->toArray()['access'];
+
+        // check for general panel access
+        if (($permissions['panel'] ?? false) !== true) {
+            throw new PermissionException(t('error.access.panel'));
+        }
+
+        // don't check if the area is not defined
+        if (empty($areaId) === true) {
+            return true;
+        }
+
+        // undefined area permissions means access
+        if (isset($permissions[$areaId]) === false) {
+            return true;
+        }
+
+        // no access
+        if ($permissions[$areaId] !== true) {
+            throw new PermissionException(t('error.access.view'));
+        }
+
+        return true;
+    }
+
+    /**
+     * Creates global data for the Panel.
+     * This will be injected in the full Panel
+     * view via the script tag.
+     *
+     * @param \Kirby\Cms\App $kirby
+     * @return array
+     */
+    public static function globals(App $kirby): array
```

Comment:
Why do we have both the shared data in the `data()` method and this `globals()` method?

Reply:
Some shared data should be renewed on navigation, some globale can stay stable for the whole session (only passed on hard reload).

Reply:
Ah, I see. Maybe it's already enough to update the method description for `globals()` to explain this.

Reply:
I extended the method description